### PR TITLE
Add subdivisions to PlaneMeshBuilder

### PR DESCRIPTION
**Pull request by IceSentry**
# Objective

- Plane subdivision was removed without providing an alternative

## Solution

- Add subdivision to the PlaneMeshBuilder

---

## Migration Guide

If you were using `Plane` `subdivisions`, you now need to use `Plane3d::default().mesh().subdivisions(10)`

fixes https://github.com/bevyengine/bevy/issues/13258

### CODE REVIEW

**Review by NiseVoid — APPROVED** [review]

File: crates/bevy_render/src/mesh/primitives/dim3/plane.rs [hunk 1]
```diff
@@ -65,27 +78,66 @@ impl PlaneMeshBuilder {
...
+                let tx = x as f32 / (x_vertex_count - 1) as f32;
+                let tz = z as f32 / (z_vertex_count - 1) as f32;
+                let pos = rotation * Vec3::new((-0.5 + tx) * size.x, 0.0, (-0.5 + tz) * size.y);
+                positions.push(pos);
+                normals.push(self.plane.normal.to_array());
+                uvs.push([tx, tz]);
+            }
+        }
+
+        for y in 0..z_vertex_count - 1 {
```
Comment (mweatherley) [inline crates/bevy_render/src/mesh/primitives/dim3/plane.rs:124]:
```suggestion
        for z in 0..z_vertex_count - 1 {
```
Maybe call this `z` as above?

File: crates/bevy_render/src/mesh/primitives/dim3/plane.rs [hunk 2]
```diff
@@ -65,27 +78,66 @@ impl PlaneMeshBuilder {
...
+            for x in 0..x_vertex_count - 1 {
+                let quad = y * x_vertex_count + x;
```
Comment (mweatherley) [inline crates/bevy_render/src/mesh/primitives/dim3/plane.rs:126]: A little comment or something about what `quad` represents and how it's being used would be nice; I think this is the only part of this implementation that isn't basically obvious

**Review by mweatherley — APPROVED** [review]

**Review by Olle-Lukowski — APPROVED** [review]

File: crates/bevy_render/src/mesh/primitives/dim3/plane.rs (hunk 2 above)
Comment (IceSentry) [reply crates/bevy_render/src/mesh/primitives/dim3/plane.rs]: that code was copied from the original bevy impl I'm not entirely sure why it was named like that but I just copied it.

Comment (mweatherley) [reply crates/bevy_render/src/mesh/primitives/dim3/plane.rs]: That makes a lot of sense :)

**Pull request merged**